cwms-data-api/src/main/java/cwms/cda/helpers/ReplaceUtils.java

```diff
@@ -38,15 +38,19 @@ public static String replace(@NotNull String template, @NotNull String key, Stri
      * @return the modified string template with the key replaced by the value
      * @throws NullPointerException if the template is null
      */
-    public static String replace(@NotNull String template, @NotNull String key, String value, boolean encode){
+    public static String replace(@NotNull String template, String key, String value, boolean encode){
```

Comment:
What ends up calling this with key==null ?  Having this handle null does seem clean so I'm not opposed its just weird that something wants to do replacement and doesn't know the key its replacing

Reply:
I use the same pattern for getAll and getOne. In the case of getAll we have a template since we don't know which row spec, but in getOne we know the row spec. I could template them both but that seemed confusing to me too.